crates/services/upgradable-executor/src/executor.rs

```diff
@@ -383,7 +386,12 @@ where
                     self.native_produce_inner(block, options, dry_run)
                 }
                 ExecutionStrategy::Wasm { module } => {
-                    self.wasm_produce_inner(module, block, options, dry_run)
+                    let maybe_blocks_module = self.get_module(block_version).ok();
+                    if let Some(blocks_module) = maybe_blocks_module {
+                        self.wasm_produce_inner(&blocks_module, block, options, dry_run)
+                    } else {
+                        self.wasm_produce_inner(module, block, options, dry_run)
+                    }
```

Comment:
question: is this needed in the case that the node does not have the most recent version of the bytecode available, hence it will fallback to the one specified in the `ExecutionStrategy::Wasm` value?

Reply:
It is for the case of the `FUEL_ALWAYS_USE_WASM=true`. We use it in the cases of tests, benchmarks, or if you just want to run WASM executor instead of the native execution.

The ideas of this change is to just use the write state transition instead of WASM version of the native state transition logic, since it can be different if you compiled the binary with different list of feature(it affects the native execution code and the WASM version as well).